### PR TITLE
mayhem

### PR DESCRIPTION
**Pull request by zwimer**

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
```
Comment (xansec) [inline .github/dependabot.yml:1]: Can you comment on why this file was added?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+RUN make -j$(nproc)
+
+RUN mkdir -p /fuzz/in
+RUN mkdir -p /fuzz/out
+RUN mkdir -p /fuzz/mount
+COPY smol.img /fuzz/in
+
+WORKDIR /fuzz
+ENTRYPOINT ["afl-fuzz", "-i", "/fuzz/in", "-o", "/fuzz/out"]
+CMD ["/ntfs-3g/src/ntfs-3g", "@@", "/fuzz/mount"]
```
Comment (xansec) [inline Dockerfile:20]: This is a Mayhemfile formatted command, and will fail if run outside of Mayhem

**Review by xansec — CHANGES_REQUESTED** [review]
Is it necessary to add `smol.img`? Despite the name, it is 1mb in size, and will likely be truncated by Mayhem anyways without explicitly allowing it. I wonder if there is a smaller corpus file you can use.

**Comment by r00tus3r** [comment]
Unfortunately, it wont be possible to use a smaller image. `ntfs-3g` is a file system driver and smallest image it can mount is 1 MB.

**Comment by xansec** [comment]
Understood. Since it's already run on Mayhem, the corpus already exists (and can be referenced in the Mayhemfile). I don't see any reason to push `smol.img` to the repo itself.

File: Mayhemfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+image: r00tus3r/mayhem-ntfs
+duration: 90
+project: mayhem-targets
```
[the remark this answers is not in the document]
Comment (r00tus3r) [reply Mayhemfile]: Done

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
+FROM aflplusplus/aflplusplus
```
[the remark this answers is not in the document]
Comment (r00tus3r) [reply Dockerfile]: Done

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ENV AFL_USE_ASAN=1
+WORKDIR /ntfs-3g
+RUN ./autogen.sh && CC=afl-clang-lto CFLAGS="-O3" CXX=afl-clang-lto++ RANLIB=llvm-ranlib-12 AR=llvm-ar-12 ./configure --enable-shared=no
+RUN make -j$(nproc)
+
+RUN mkdir -p /fuzz/in
+RUN mkdir -p /fuzz/out
+RUN mkdir -p /fuzz/mount
+COPY smol.img /fuzz/in
+
```
[the remark this answers is not in the document]
Comment (r00tus3r) [reply Dockerfile:13]: Done

File: .github/dependabot.yml (hunk 1 above)
Comment (r00tus3r) [reply .github/dependabot.yml]: Removed, it was added by mistake

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+WORKDIR /
+RUN git clone https://github.com/tuxera/ntfs-3g.git
+
+ENV AFL_USE_ASAN=1
+WORKDIR /ntfs-3g
+RUN ./autogen.sh && CC=afl-clang-lto CFLAGS="-O3" CXX=afl-clang-lto++ RANLIB=llvm-ranlib-12 AR=llvm-ar-12 ./configure --enable-shared=no
+RUN make -j$(nproc)
+
+RUN mkdir -p /fuzz/in
```
[the remark this answers is not in the document]
Comment (r00tus3r) [reply Dockerfile]: Okay, updated

File: Dockerfile (hunk 2 above)
Comment (r00tus3r) [reply Dockerfile]: Updated

**Comment by r00tus3r** [comment]
@xansec my changes in the dockerfile does not seem to be working on the mayhem server, locally it is working. And I'm not getting any useful logs from mayhem. Would you be able to take a look? Thanks!

**Comment by xansec** [comment]
I was able to get this going by bumping timeout and memory limits. But I don't think this is the most effective target. I'm willing to accept it, but we may want to look into improving the efficiency of the target.
See if adding the changes I made to this run help at all:
https://mayhem.forallsecure.com/zwimer/ntfs-3g/ntfs-3g/5

**Pull request merged**